components/push/src/communications.rs

```diff
@@ -411,7 +397,16 @@ impl Connection for ConnectHttp {
             return Ok(false);
         }
         let local_channels: HashSet<String> = channels.iter().cloned().collect();
-        let remote_channels: HashSet<String> = HashSet::from_iter(self.channel_list()?);
+        let remote_channels: HashSet<String> = match self.channel_list() {
+            Ok(v) => HashSet::from_iter(v),
+            Err(e) => match e.kind() {
+                UAIDNotRecognizedError(_) => {
+                    self.uaid = None;
```

Comment:
Do we also want to call `unsubscribe` in this case, like we do for when there's a mismatch in the channel lists? (I note that `unsubscribe` contains its own internal version of setting `uaid = None`).

Reply:
Hmmm in this case I don't think we should call `unsubscribe`, because we only get this error if the server lost our UAID for one reason or the other, and the `unsubscribe` network request would just fail with the same error...

That said, I think it's begging for a comment to describe what I just said

Reply:
this raises an interesting question that probably requires a decent bit of high-level figuring out on my end....

if the server loses our UAID without us explicitly `unsubscribe`ing, how bad is it on the server, and is it really OK for the client to request another UAID as opposed to erroring out? I'm assuming the client will eventually request another UAID after the verifyConnection is returned to the consumer, and the consumer decides to resubscribe.

Like, will we have dangling connections to firebase/etc, and if we create a second connection, how bad is it? I'll need to look more into what specific cases do we end up losing the UAID (I'm assuming there are more cases than the fix I did on #4325). if we lose the UAID because we called `unsubscribe(None)` this is fine. If we lose it for some other reason, we might want to check and confirm that requesting another UAID doesn't leave our overall system in a bad state

anyways, I might be waaay overthinking things 😅

Reply:
>  and the unsubscribe network request would just fail with the same error...

Ah yes, I see - in my head the `unsubscribe` call was also deleting the local records but that's not the case, that happens at a higher level of the call stack. Thanks!

Reply:
> I'm assuming the client will eventually request another UAID after the verifyConnection is returned to the consumer,
> and the consumer decides to resubscribe.

I believe that's the intention, yes.

IIUC what's supposed to happen if there is a server-side error, is that the native push layer signals to the app that it needs to reconnect, and the app [registers a new token](https://github.com/mozilla/application-services/blob/eae2a401008e155bb6ebcb3375b77db70a19fd9d/components/push/src/subscriber.rs#L128) with the autopush server. However, I don't think we have a good shared understanding of all the possible combinations of (client state, autopush state, native push infra state) and the possible transitions between them. A valuable longer-term effort here would be to construct a state transition diagram for each of the three systems interacting here, it will probably surface some interactions we haven't thought of yet.

Reply:
filed #4373!